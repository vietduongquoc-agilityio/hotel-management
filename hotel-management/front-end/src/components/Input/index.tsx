import { forwardRef, memo } from "react";
import { Input as InputUI, InputProps } from "@chakra-ui/react";
import { getStyleInput } from "./InputStyle";

export type InputType = "first" | "second";

interface InputTypeProps extends InputProps {
  placeHolder: string;
  inputType: InputType;
}

const Input = forwardRef<HTMLInputElement, InputTypeProps>(
  ({ placeHolder, inputType, ...restProps }, ref) => {
    const styleFirst = getStyleInput(inputType);

    const propsStyle = {
      ...styleFirst,
      placeholder: placeHolder,
      ...restProps,
    };

    return <InputUI ref={ref} {...propsStyle} />;
  }
);

Input.displayName = "Input";

export default memo(Input);
