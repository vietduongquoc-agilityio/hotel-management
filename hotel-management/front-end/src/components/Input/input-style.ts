import { InputType } from ".";

export const getStyleInput = (type: InputType) => {
  switch (type) {
    case "primary": {
      return primary;
    }
    case "number": {
      return number;
    }
  }
};

const primary = {
  size: "md",
  variant: "outline",
};

const number = {
  size: "md",
  variant: "outline",
};
