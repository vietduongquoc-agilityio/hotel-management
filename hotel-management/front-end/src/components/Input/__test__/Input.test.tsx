import React from "react";
import { render, screen } from "@testing-library/react";
import Input from "../index";
import { getStyleInput } from "../InputStyle";

jest.mock("../InputStyle", () => ({
  getStyleInput: jest.fn(),
}));

describe("Input Component", () => {
  it("should render with the correct placeholder", () => {
    render(<Input placeHolder="Test Placeholder" inputType="default" />);
    const inputElement = screen.getByPlaceholderText("Test Placeholder");
    expect(inputElement).toBeInTheDocument();
  });

  it("should apply styles based on the inputType", () => {
    const mockStyle = { size: "md", variant: "outline" };
    (getStyleInput as jest.Mock).mockReturnValue(mockStyle);

    render(<Input placeHolder="Styled Input" inputType="default" />);
    const inputElement = screen.getByPlaceholderText("Styled Input");

    // Assert that the getStyleInput function was called with the correct inputType
    expect(getStyleInput).toHaveBeenCalledWith("first");

    // Ensure that Chakra UI's Input received the correct styles
    expect(inputElement).toHaveAttribute("size", mockStyle.size);
    expect(inputElement).toHaveAttribute("variant", mockStyle.variant);
  });

  it("should pass other props to the Chakra Input", () => {
    render(
      <Input
        placeHolder="Extra Props Test"
        inputType="default"
        data-testid="custom-input"
        isDisabled
      />
    );

    const inputElement = screen.getByTestId("custom-input");
    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toBeDisabled();
  });
});
