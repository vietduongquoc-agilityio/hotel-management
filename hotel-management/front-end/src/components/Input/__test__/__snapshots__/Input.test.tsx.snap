// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should render with the correct placeholder 1`] = `
<div>
  <input
    class="chakra-input css-0"
    placeholder="Test Placeholder"
  />
</div>
`;
